networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -0,0 +1,255 @@
+# Copyright 2014 Cisco Systems, Inc.  All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import re
+
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from neutron.common import constants as l3_constants
+from neutron import manager
+from neutron.plugins.common import constants as svc_constants
+
+from networking_cisco.plugins.cisco.device_manager.plugging_drivers import (
+    hw_vlan_trunking_driver as hw_vlan)
+
+LOG = logging.getLogger(__name__)
+
+APIC_OWNED = 'apic_owned_'
+UUID_REGEX = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
+DEVICE_OWNER_ROUTER_GW = l3_constants.DEVICE_OWNER_ROUTER_GW
+DEVICE_OWNER_ROUTER_INTF = l3_constants.DEVICE_OWNER_ROUTER_INTF
+
+ACI_ASR1K_DRIVER_OPTS = [
+    cfg.StrOpt('aci_transit_nets_config_file', default=None,
+               help=_("ACI with ASR transit network configuration file.")),
+]
+
+cfg.CONF.register_opts(ACI_ASR1K_DRIVER_OPTS, "general")
+
+DEFAULT_EXT_DICT = {'gateway': '1.103.2.1',
+                    'next_hop': '1.103.2.254',
+                    'host_pool_cidr': '1.103.2.0/24',
```

Comment:
In our neutron l3out config section, host_pool_cidr means the SNAT subnet and it doesn't have to be in the same subnet of gateway/next_hop. We should name it differently if it means something else here.